### PR TITLE
 flat `Owners` file, support owners per folder 

### PR DESCRIPTION
**Pull request by myakove**
Each folder can have its own OWNERS file to set approvers and reviewers for the folder.

## Release Notes

- **New Features**
  - Enhanced error handling for webhook processing, providing clearer logging of exceptions.
  - Improved retrieval of `OWNERS` files with added flexibility and error handling.
  - Expanded support for multiple Python development environments in the container.
  - Added methods for aggregating approvers and reviewers from `OWNERS` files.
  - Introduced logging setup and automatic webhook creation for GitHub repositories.

- **Documentation**
  - Updated `README.md` to clarify functionality and usage of the GitHub webhook server, including detailed instructions and configuration examples.

- **Configuration Updates**
  - Updated Python version requirement to 3.10 and above in `pyproject.toml`.
  - Added new development dependencies for improved development experience.

- **Bug Fixes**
  - Corrected minor typographical errors in documentation for better clarity.

### CODE REVIEW

File: Dockerfile
Comment (myakove) [reply Dockerfile:23]: This is to use inside the container for others tasks that required python

File: pyproject.toml
Comment (myakove) [reply pyproject.toml:41]: ack

File: webhook_server_container/libs/github_api.py
Comment (myakove) [reply webhook_server_container/libs/github_api.py]: fixed

File: webhook_server_container/libs/github_api.py
Comment (myakove) [reply webhook_server_container/libs/github_api.py:604]: this is handle in `assign_reviewers`

**Comment by myakove** [comment]
Fix https://github.com/myk-org/github-webhook-server/issues/621

File: webhook_server_container/libs/github_api.py
Comment (dbasunag) [inline webhook_server_container/libs/github_api.py:620]: Say the PR changes file in  folder 1 and folder 2, both with different OWNERS and under root folder. In this case only approvers should be their parent(root's) OWNERS file.

File: webhook_server_container/libs/github_api.py
Comment (dbasunag) [inline webhook_server_container/libs/github_api.py:2066]: You can limit with folder level too. That way you can limit what a SIG can be (upto level 3 - root>folder level1> folderlevel 2) etc.

Comment (myakove) [reply webhook_server_container/libs/github_api.py:2067]: ANy reason to limit this?

File: webhook_server_container/libs/github_api.py
Comment (myakove) [reply webhook_server_container/libs/github_api.py]: I don't understand what you mean.

File: webhook_server_container/libs/github_api.py
Comment (myakove) [reply webhook_server_container/libs/github_api.py:2101]: I don't want to put this in the config

**Comment by myakove** [comment]
@coderabbitai pause

**Review by dbasunag — APPROVED** [review]

File: webhook_server_container/libs/github_api.py
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:2090]: Data on SIG is all over the place. Initially based on what I was reading, I was thinking a SIG has to be large enough to justify forming it. If that is true then level 3/4 would be good enough. But then I found staging/src/kubevirt.io/api/OWNERS and while it is one instance, since this repo is not just managing one repo, you can't predict or restrict levels. So taking that comment back.

**Review by dbasunag — APPROVED** [review]

File: webhook_server_container/libs/github_api.py
Comment (dbasunag) [inline webhook_server_container/libs/github_api.py:2147]: I am a bit confused here. Looking at this code, if this is the folder structure:
```
-root (owner0, approver 0 and reviewer 0) 
         -|-folder1(owner1 (approver 1 and reviewer 1)) - file1, file2, file3
          | -folder2(owner2 approver 2 and reviewer2)) - file 4, file 5)
          |-folder 3
```
if a patch touches folder 1 and folder 3, then the reviewers and approvers would be approver 0, approver 1, reviewer 0, reviewer1 is this assumption right?
While for reviewers everyone should be included, for approvers, I would need more restrictive approach. e.g. if the any of the changed_folder_path, is not fully under a SIG's ownership, the approval should be based on OWNER file under the first common ancestor. I think if we can talk I can explain it better.

File: webhook_server_container/tests/test_github_api.py
Comment (dbasunag) [inline webhook_server_container/tests/test_github_api.py:184]: You might want to add a test for missing approval from 1 and 2.

**Review by dbasunag — APPROVED** [review]
Thank you. Looks good.

**Review by rnetser — APPROVED** [review]

**Pull request merged**
